### PR TITLE
BUILD-3774: add new CI pipelines for Connector JS

### PR DESCRIPTION
**Pull request by ManuelJNunez**

### CODE REVIEW

**Comment by ManuelJNunez** [comment]
Pending on this PR: https://github.com/rticommunity/rticonnextdds-connector/pull/87

File: package.json [hunk 1]
```diff
@@ -40,6 +42,8 @@
   "devDependencies": {
     "chai": "^4.3.4",
     "chai-as-promised": "^7.1.1",
+    "mocha": "^10.2.0",
```
Comment (ManuelJNunez) [inline package.json:45]: Added mocha here as devDependency

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: Build & Test Connector JS
+
+    outputs:
+      pr_number: ${{ steps.pr_number.outputs.PR_NUMBER }}
+
+    steps:
```
Comment (ManuelJNunez) [inline .github/workflows/build.yaml:29]: Firsts steps a comment-pr job were copied from [here](https://github.com/ManuelJNunez/rticonnextdds-examples/blob/master/.github/workflows/build.yml).

File: .gitmodules [hunk 3]
```diff
@@ -1,3 +1,4 @@
 [submodule "rticonnextdds-connector"]
 	path = rticonnextdds-connector
 	url = https://github.com/rticommunity/rticonnextdds-connector.git
+	branch = develop
```
Comment (samuelraeburn) [inline .gitmodules:4]: Does this mean it will always checkout develop of the remote?

Comment (ManuelJNunez) [reply .gitmodules:4]: Yes, take into account each branch will have a different configuration in the `config.yaml` file.

**Review by samuelraeburn — APPROVED** [review]

**Review by lulivi — APPROVED** [review]

**Pull request merged**